ocs_ci/ocs/ui/block_pool.py

```diff
@@ -0,0 +1,174 @@
+        """
+        Check if pool appears in the block pool list
+
+        Args:
+            pool_name (str): Name of the pool to check
+
+        Return:
+            bool: True if pool is in the list of pools page, otherwise False
+
+        """
+        self.navigate_overview_page()
+        self.navigate_block_pool_page()
+        self.page_has_loaded()
+        pool_existence = self.wait_until_expected_text_is_found(
+            (f"a[data-test-operand-link={pool_name}]", By.CSS_SELECTOR), pool_name, 5
+        )
+        logger.info(f"Pool name {pool_name} existence is {pool_existence}")
+        return pool_existence
+
+    def delete_pool(self, pool_name):
+        """
+        Delete pool from pool page
+
+        Args:
+            pool_name (str): The name of the pool to be deleted
+
+        Returns:
+            bool: True if pool is not found in pool list, otherwise false
+
+        """
+
+        self.navigate_overview_page()
+        self.navigate_block_pool_page()
+        self.page_has_loaded()
+        self.do_click((f"{pool_name}", By.LINK_TEXT))
+        self.do_click(self.bp_loc["actions_inside_pool"])
+        self.do_click(self.bp_loc["delete_pool_inside_pool"])
+        self.do_click(self.bp_loc["confirm_delete_inside_pool"])
+        # wait for pool to deleted
+        time.sleep(2)
+        return not self.check_pool_existence(pool_name)
+
+    def edit_pool_parameters(self, pool_name, replica=3, compression=True):
+        """
+        Edit an already existing pool
+
+        Args:
+            pool_name (str): The name of the pool to change.
+            replica (int): size of replica. Available in OCS now 2,3.
+            compression (bool): True if enable compression. False otherwise.
+
+        """
+        self.navigate_overview_page()
+        self.navigate_block_pool_page()
+        self.page_has_loaded()
+        self.do_click([f"{pool_name}", By.LINK_TEXT])
+        self.do_click(self.bp_loc["actions_inside_pool"])
+        self.do_click(self.bp_loc["edit_pool_inside_pool"])
+        self.do_click(self.bp_loc["replica_dropdown_edit"])
+        if replica == 2:
+            self.do_click(self.bp_loc["second_select_replica_2"])
+        else:
+            self.do_click(self.bp_loc["second_select_replica_3"])
+        compression_checkbox_status = self.get_checkbox_status(
+            self.bp_loc["compression_checkbox_edit"]
+        )
+        if compression != compression_checkbox_status:
+            self.do_click(self.bp_loc["compression_checkbox_edit"])
+        self.do_click(self.bp_loc["save_pool_edit"])
+
+    def reach_pool_limit(self, replica, compression):
+        """
+        Add pools till pool fails because of pg limit.
+
+        Args:
+             replica (int): size of pool.
+             compression (bool): True for enabling compression. Otherwise False.
+
+        """
+        pool_list = []
+        ceph_pod = pod.get_ceph_tools_pod()
+
+        while True:
```

Comment:
Is it possible not to use `while True`? maybe generator?
It's dangerous to use `while True`.. for example, `pool_status` is `True` every time...

Reply:
I doubt that pool status will be forever true because it generates a pool every iteration and you can create pools till pg limit is reached which is like 16-20 pools max.
Anyway there is something in your saying but a generator is an overkill. I've added a counter with a limit of 50.